resources/assert_lib.brs

```diff
@@ -60,3 +61,81 @@ function __formatError(errMessage)
         stack: {}
     }
 end function
+
+sub __deepEquals(actual, expected, error)
+    if deepEquals(actual, expected) = true then
+        m.__pass()
+    else
+        m.__fail(m.formatError(error))
+    end if
+end sub
+
+'utility function to compare two objects/associative arrays with nested data structures
+function deepEquals(left as object, right as object) as boolean
+    ' because Brightsript does not support closures and we are doing recursion,
+    'we want to hold the function parameters in top-level state variables during each execution
+    updatedLeftState = invalid
+    updatedRightState = invalid
+    isEqual = true
+
+    if type(left) <> type(right) then return false
```

Comment:
`Left` and `Right` are [global string functions](https://developer.roku.com/en-gb/docs/references/brightscript/language/global-string-functions.md#lefts-as-string-n-as-integer-as-string) that return substrings of an input value.  I'm pretty sure the `left` and `right` variables here shadow those global variables, but if that behavior ever changes in the reference implementation this will be a tricky bug to find!  Would you mind calling these parameters `a` and `b` or something like that to avoid confusion with the global functions called `left` and `right`?